AEPMessaging/Sources/Messaging+EdgeEvents.swift

```diff
@@ -0,0 +1,220 @@
+            return xdmDict
+        }
+
+        // required keys are found using the following priority
+        // 1. check the event's "mixins" key
+        // 2. check the event's "cjm" key
+        var mixins: [String: Any]
+        if event.mixins != nil {
+            mixins = event.mixins!
+        } else {
+            guard let cjm = event.cjm else {
+                Log.warning(label: MessagingConstants.LOG_TAG,
+                            "Failed to update xdmMap with Adobe/AJO information : Adobe/AJO data is not available in the event '\(event.id.uuidString)'.")
+                return xdmDict
+            }
+
+            mixins = cjm
+        }
+
+        var xdmDictResult = xdmDict
+
+        // Add all the key and value pair to xdmDictResult
+        xdmDictResult.mergeXdm(rhs: mixins)
+
+        // Check if the xdm data provided by the customer is using cjm for tracking
+        // Check if both `MessagingConstant.AdobeTrackingKeys.EXPERIENCE` and `MessagingConstant.AdobeTrackingKeys.CUSTOMER_JOURNEY_MANAGEMENT` exists
+        if var experienceDict = xdmDictResult[MessagingConstants.AdobeTrackingKeys.EXPERIENCE] as? [String: Any] {
+            if var cjmDict = experienceDict[MessagingConstants.AdobeTrackingKeys.CUSTOMER_JOURNEY_MANAGEMENT] as? [String: Any] {
+                // Adding Message profile and push channel context to CUSTOMER_JOURNEY_MANAGEMENT
+                guard let messageProfile = MessagingConstants.AdobeTrackingKeys.MESSAGE_PROFILE_JSON.toJsonDictionary() else {
+                    Log.warning(label: MessagingConstants.LOG_TAG,
+                                "Failed to update xdmMap with adobe/cjm informations:" +
+                                    "converting message profile string to dictionary failed in the event '\(event.id.uuidString)'.")
+                    return xdmDictResult
+                }
+                // Merging the dictionary
+                cjmDict.mergeXdm(rhs: messageProfile)
+                experienceDict[MessagingConstants.AdobeTrackingKeys.CUSTOMER_JOURNEY_MANAGEMENT] = cjmDict
+                xdmDictResult[MessagingConstants.AdobeTrackingKeys.EXPERIENCE] = experienceDict
+            }
+        } else {
+            Log.warning(label: MessagingConstants.LOG_TAG,
+                        "Failed to send adobe/cjm information data with the tracking," +
+                            "\(MessagingConstants.AdobeTrackingKeys.EXPERIENCE) is missing in the event '\(event.id.uuidString)'.")
+        }
+        return xdmDictResult
+    }
+
+    /// Adding application data based on the application opened or not
+    /// - Parameters:
+    ///   - applicationOpened: `Bool` stating whether the application is opened or not
+    ///   - xdmData: `[AnyHashable: Any]` xdm data in which application data needs to be added
+    /// - Returns: `[String: Any]` which contains the application data
```

Comment:
- Returns: `Dictionary` which contains the application data.

Same for other returns